### PR TITLE
S3BatchDelete Edge Cases

### PR DESCRIPTION
**Pull request by allenaverbukh**
## Summary
Addresses edge cases to return status of 200 and an Error object. 
1) max batch size exceeds
2) empty objects / no objects
3) empty channel read 
4) malformed XML request body

## Testing Done
Wrote unit tests for the above cases.

### CODE REVIEW

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BatchDeleteHandler.java [hunk 1]
```diff
@@ -90,17 +93,40 @@ private Callback<Long> parseRequestBodyAndDeleteCallback(RetainingAsyncWritableC
...
+      try {
+        // check for empty message
+        if (bytesRead == 0) {
+          logger.info("bytesRead is empty");
+          throw new RestServiceException("bytesRead is empty", RestServiceErrorCode.BadRequest);
+        }
+
+        // ensure request body format is correct
+        deleteRequest = deserializeRequest(channel);
+        if (deleteRequest.getObjects() == null) {
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BatchDeleteHandler.java:106]: let do 
```
if (deleteRequest.getObjects() == null || deleteRequest.getObjects().size() == 0) {
}
```

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BatchDeleteHandler.java [hunk 2]
```diff
@@ -170,6 +192,7 @@ public S3MessagePayload.S3BatchDeleteObjects deserializeRequest(RetainingAsyncWr
       byteBuffer.readBytes(byteArray);
       return new XmlMapper().readValue(byteArray, S3MessagePayload.S3BatchDeleteObjects.class);
     } catch (Exception e) {
+      logger.info("s3batchdelete failed to deserialize request");
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BatchDeleteHandler.java:195]: let's do logger.trace.

File: ambry-frontend/src/test/java/com/github/ambry/frontend/S3BatchDeleteHandlerTest.java [hunk 3]
```diff
@@ -121,6 +122,73 @@ public void deleteObjectTest() throws Exception {
...
+    XmlMapper xmlMapper = new XmlMapper();
+    S3MessagePayload.Error response =
+        xmlMapper.readValue(byteBuffer.array(), S3MessagePayload.Error.class);
+    assertEquals(response.getMessage(), ERR_MALFORMED_REQUEST_BODY_MESSAGE);
+    assertEquals(response.getCode(), ERR_MALFORMED_REQUEST_BODY_CODE);
+    assertEquals("Mismatch on status", ResponseStatus.Ok, restResponseChannel.getStatus());
+  }
+
+  @Test
+  public void emptyRequestTest() throws Exception {
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/test/java/com/github/ambry/frontend/S3BatchDeleteHandlerTest.java:166]: let's also test the 1000 blob keys error.

**Review by justinlin-linkedin — APPROVED** [review]
